lib/web_ui/skwasm/text/paragraph.cpp

```diff
@@ -74,14 +74,13 @@ SKWASM_EXPORT size_t paragraph_getLineCount(Paragraph* paragraph) {
 
 SKWASM_EXPORT int paragraph_getLineNumberAt(Paragraph* paragraph,
                                             size_t characterIndex) {
-  return paragraph->getLineNumberAt(characterIndex);
+  return paragraph->getLineNumberAtUTF16Offset(characterIndex);
 }
 
 SKWASM_EXPORT LineMetrics* paragraph_getLineMetricsAtIndex(Paragraph* paragraph,
-                                                           size_t index) {
+                                                           size_t lineNumber) {
   auto metrics = new LineMetrics();
-  paragraph->getLineMetricsAt(index, metrics);
-  return metrics;
+  return paragraph->getLineMetricsAt(lineNumber, metrics) ? metrics : nullptr;
```

Comment:
You need to `delete metrics;` in the case where you're returning a nullptr, otherwise this will leak.

Reply:
done